cli/src/main.rs

```diff
@@ -248,96 +258,95 @@ fn generate_flowgraph(
-            })();
+        println!("{flowgraph}");
 
-            match result {
-                Ok(PromiseState::Pending) => {
-                    eprintln!("module `{}` didn't execute", file.display());
-                }
-                Ok(PromiseState::Fulfilled(_)) => {}
-                Ok(PromiseState::Rejected(err)) => {
-                    eprintln!("Uncaught {}", err.display());
-
-                    if let Ok(err) = JsError::from_opaque(err).try_native(context) {
-                        if let Some(cause) = err.cause() {
-                            eprintln!("\tCaused by: {cause}");
-                        }
-                    }
-                }
-                Err(err) => {
-                    eprintln!("Uncaught {err}");
+        return Ok(());
+    }
 
-                    if let Ok(err) = err.try_native(context) {
-                        if let Some(cause) = err.cause() {
-                            eprintln!("\tCaused by: {cause}");
-                        }
-                    }
-                }
-            }
-        } else {
-            match context.eval(Source::from_bytes(&buffer)) {
-                Ok(v) => println!("{}", v.display()),
-                Err(v) => eprintln!("Uncaught {v}"),
+    if args.module {
+        let module = Module::parse(Source::from_filepath(file)?, None, context)
+            .map_err(|e| e.into_erased(context))?;
+
+        loader.insert(
+            file.canonicalize()
+                .wrap_err("could not canonicalize input file path")?,
+            module.clone(),
+        );
+
+        let promise = module.load_link_evaluate(context);
+        context.run_jobs();
+        let result = promise.state();
+
+        return match result {
+            PromiseState::Pending => Err(eyre!("module didn't execute")),
+            PromiseState::Fulfilled(_) => Ok(()),
+            PromiseState::Rejected(err) => {
+                return Err(JsError::from_opaque(err).into_erased(context).into())
             }
-            context.run_jobs();
-        }
+        };
+    }
+
+    match context.eval(Source::from_filepath(file)?) {
+        Ok(v) => println!("{}", v.display()),
+        Err(v) => eprintln!("Uncaught {v}"),
     }
+    context.run_jobs();
 
     Ok(())
 }
 
-fn main() -> Result<(), io::Error> {
+fn evaluate_files(args: &Opt, context: &mut Context, loader: &SimpleModuleLoader) {
+    for file in &args.files {
+        let Err(err) = evaluate_file(file, args, context, loader)
+            .wrap_err_with(|| eyre!("could not evaluate file `{}`", file.display()))
+        else {
+            continue;
+        };
+
+        eprintln!("{err:?}");
+    }
+}
+
+fn main() -> Result<()> {
+    color_eyre::config::HookBuilder::default()
+        .display_location_section(false)
+        .display_env_section(false)
+        .install()?;
+
+    if std::env::var("RUST_BACKTRACE").is_err() {
+        std::env::set_var("RUST_BACKTRACE", "0");
+    }
```

Comment:
Why do we need to do this?

Reply:
Forgot to remove it. It was an attempt to remove the error location which we don't need for a public CLI, but it happened to be configured by using the `HookBuilder`.